conda/base/context.py

```diff
@@ -489,6 +496,11 @@ def __init__(self, search_path=None, argparse_args=None, **kwargs):
         self._set_env_vars(APP_NAME)
         self._set_argparse_args(argparse_args)
 
+        # set plugin configuration parameters
+        if initialize_plugin_config:
+            self.plugin_manager.load_settings()
+            self.plugins = PluginConfig(context.raw_data)
+
```

Comment:
I wouldn't be opposed to this. I had to do this because I was running into lots of circular import errors.

We could even do something like the following to lazy load the `plugins` property itself:

```python
def __init__(self, search_path=None, argparse_args=None, plugin_config=None):
    # snip, snip...

    # This is just here in case we ever want to override what `plugins` is on object instantiation
    if plugin_config is not None:
        self._plugins = plugin_config


@property
def plugins():
    if not self._plugins:
        self.plugins = PluginConfig(context.raw_data)
    return self._plugins
```